src/types/player.ts

```diff
@@ -161,8 +159,28 @@ export class SelfPlayer extends Player {
         }})
     }
 
-    public say(content: string) {
-        return this.client.send(Magic(0x6B), Bit7(MessageType['chatMessage']), String(content))
+    public say(content: string): void;
+    public say(preamble: string, content: string): void;
+    public say(preamble: string, content?: string) {
+        if (content == undefined) {
+            return this.say(this.client.chatPrefix || '', preamble)
+        }
+
+        const MESSAGE_SIZE = 120
+        const CONTENT_ALLOWED_SIZE = MESSAGE_SIZE - preamble.length - (preamble.length > 0 ? 1 : 0)
+
+        if (preamble.length > MESSAGE_SIZE)
+            throw new Error('Chat preamble is larger than message size. Bad.')
+
+        // TODO regex?
+        if (content.length > CONTENT_ALLOWED_SIZE) {
+            const separator = content.substring(0, CONTENT_ALLOWED_SIZE).lastIndexOf(' ') || CONTENT_ALLOWED_SIZE
+            this.say(preamble, content.substring(0, separator))
+            this.say(preamble, content.substring(separator, content.length))
+            return
+        }
```

Comment:
@ScorpioneOrzion This is what currently is happening, will rewrite to regex later